File: Data Structures & Algorithms/decode-string/submission-0.cpp
class Solution {
public:
    string decodeString(string s) {
        stack<string> stk;
        for(auto c: s) {
            if(c != ']') {
                stk.push(string(1, c));
            } else {
                string sub = "";
                while(stk.top()!="[") {
                    sub = stk.top() + sub;
                    stk.pop();
                }
                stk.pop();
                string num = "";
                while(stk.size() && isdigit(stk.top()[0])) {
                    num = stk.top() + num;
                    stk.pop();
                }

                int rep = stoi(num);

                string tmp = "";
                for(int i =0;i<rep;i++){
                    tmp = tmp+sub;
                }
                stk.push(tmp);
            }
        }
        string ans = "";

        while(stk.size()){
            ans = stk.top()+ ans;
            stk.pop();
        }
        return ans;
    }
};

File: Data Structures & Algorithms/reverse-string/submission-0.java
class Solution {
    public void reverseString(char[] s) {
        int n = s.length;
        int l = 0, r = n-1;
        while(l<r) {
            char tmp = s[l];
            s[l] = s[r];
            s[r] = tmp;
            l++;
            r--;
        }

        // return s;
    }
}